### ISSUE TITLE
Dropout layers are not activated when predicting uncertainty in some models

### ISSUE THREAD

**Issue by jc32173** [deepchem/deepchem#3115]
## 🐛 Bug

Dropout layers are not activated when `predict_uncertainty` is called in some models, so the epistemic component of the uncertainty is not calculated.  I think the models based on KerasModel and TorchModel which support uncertainty prediction are affected, e.g. GraphConv, DAG.

When the model is called in `_predict` during the loop over dropout masks in `predict_uncertainty`, it is called with `training=False` (KerasModel) or after `model.eval()` (TorchModel) which switch off the dropouts.  This means that the values of `results` calculated in each iteration of the loop are identical.

The MultitaskRegressor model used in the uncertainty tutorial is not affected by this issue though.

I can try and submit a PR to address this issue if people agree?

**Comment by peastman** [comment]
The models that support uncertainty don't rely on the `training` flag for dropout.  Instead they use an input argument to turn it on or off.  For example, in MultitaskRegressor:

https://github.com/deepchem/deepchem/blob/bf23ed30d5399d655ea828e5a45d97327271d9d6/deepchem/models/fcnet.py#L310-L311

When you call `predict_uncertainty()`, TorchModel passes `mode='uncertainty'` to the default generator.  The implementation of it for MultitaskRegressor enables the dropout flag in that case:

https://github.com/deepchem/deepchem/blob/bf23ed30d5399d655ea828e5a45d97327271d9d6/deepchem/models/fcnet.py#L384-L387

**Comment by jc32173** [comment]
That works for the MultitaskRegressor, so this issue does not affect that model.

However, I thought the GraphConv and DAG KerasModels are also meant to support uncertainty prediction?  Both models have an `uncertainty` argument and have a variance output type,

e.g. for GraphConv:

https://github.com/deepchem/deepchem/blob/bf23ed30d5399d655ea828e5a45d97327271d9d6/deepchem/models/graph_models.py#L973-L974

and for DAG:

https://github.com/deepchem/deepchem/blob/bf23ed30d5399d655ea828e5a45d97327271d9d6/deepchem/models/graph_models.py#L688-L692

As far as I can see there is no input argument to turn dropouts on during `predict_uncertainty()` for these models?
